src/command/FileCommands.cxx

```diff
@@ -224,12 +230,212 @@ handle_read_comments(Client &client, Request args, Response &r)
 	case LocatedUri::Type::ABSOLUTE:
 		return read_stream_comments(r, located_uri.canonical_uri);
 
-	case LocatedUri::Type::RELATIVE:
-		return read_db_comments(client, r, located_uri.canonical_uri);
 
-	case LocatedUri::Type::PATH:
-		return read_file_comments(r, located_uri.path);
-	}
+    case LocatedUri::Type::RELATIVE:
+        return read_db_comments(client, r, located_uri.canonical_uri);
+
+    case LocatedUri::Type::PATH:
+        return read_file_comments(r, located_uri.path);
+    }
+
+    gcc_unreachable();
+}
+
+InputStreamPtr
+find_stream_art(const char *directory)
+{
+    static const std::string art_names[] = {
+        "cover.png",
+        "cover.jpg",
+        "cover.tiff",
+        "cover.bmp"
+    };
+    
+    for(const std::string& name: art_names) {
+        const std::string art_file = PathTraitsUTF8::Build(directory, name.c_str());
+        
+        try {
+            Mutex mutex;
+            Cond cond;
+            
+            auto is = InputStream::OpenReady(art_file.c_str(), mutex, cond);
+            return is;
+        } catch (const std::exception &e) {
+            continue;
+        }
+    }
+    return nullptr;
+}
+
+static CommandResult
+read_stream_art(Response &r, const char *uri, uint64_t offset)
+{
+    std::string art_directory = PathTraitsUTF8::GetParent(uri);
+    InputStreamPtr is = find_stream_art(art_directory.c_str());
+    
+    if (is == nullptr) {
+        r.Error(ACK_ERROR_NO_EXIST, "No art file exists");
+        return CommandResult::ERROR;
+    }
+    
+    if (!is->IsSeekable()) {
+        r.Error(ACK_ERROR_NO_EXIST, "Cannot seek art file stream");
+        return CommandResult::ERROR;
+    }
+    if (!is->KnownSize()) {
+        r.Error(ACK_ERROR_NO_EXIST, "Cannot get size for art file stream");
+        return CommandResult::ERROR;
+    }
+    
+    const size_t art_file_size = is->GetSize();
+    
+    constexpr size_t CHUNK_SIZE = 8192;
+    uint8_t buffer[CHUNK_SIZE];
+    uint32_t read_size;
+    try {
+        is->Seek(offset);
+        read_size = is->Read(&buffer, CHUNK_SIZE);
+    } catch (const std::exception &e) {
+        PrintError(r, std::current_exception());
+        return CommandResult::ERROR;
+    }
+    
+    r.Format("size: %" PRIu64 "\n"
+             "file: %s\n"
+             "offset: %" PRIu64 "\n"
```

Comment:
I'd disagree that this means the client is not well-designed, but I do store the request in mine and match it with the response, so removing this wouldn't effect my client. 

However I think it's reasonable for a binary protocol to fully describe what it's sending, so would favour keeping the file name and offset in the response. The server knows both of these to load the chunk, and the overhead for 8KB chunks is in the order of 0.5%.